UniAtHome/UniAtHome.WebAPI/Controllers/UsersController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using UniAtHome.BLL.Interfaces;
+using UniAtHome.DAL.Entities;
+using UniAtHome.WebAPI.Models.UsersRequests;
+
+namespace UniAtHome.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly UserManager<User> userManager;
+
+        private readonly IAuthTokenGenerator tokenGenerator;
+
+        public UsersController(UserManager<User> userManager, IAuthTokenGenerator tokenGenerator)
+        {
+            this.userManager = userManager;
+            this.tokenGenerator = tokenGenerator;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<object> Register([FromBody] RegistrationRequest registerModel)
+        {
+            var user = new User
+            {
+                UserName = registerModel.Email,
+                FirstName = registerModel.FirstName,
+                LastName = registerModel.LastName,
+                Email = registerModel.Email
+            };
+            var registerResult = await userManager.CreateAsync(user, registerModel.Password);
```

Comment:
Please, don't use UserManager here. We should use Services and Repositories.

Reply:
Shall I encapsulate UserManager inside UsersRepository? However, in order to create the Repository I'll need UserManager dependency. There is no way to remove it.
Or do I need to recreate the whole UserManager functions from scratch including password verification, adding roles and claims etc?
Or can I extend UserManager by Repository?